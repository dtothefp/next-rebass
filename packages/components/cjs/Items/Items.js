"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _socket = _interopRequireDefault(require("socket.io-client"));

var _redux = require("@css/redux");

var _rebass = require("rebass");

var _ItemsNav = _interopRequireDefault(require("../ItemsNav/ItemsNav"));

var _Item = _interopRequireDefault(require("../Item/Item"));

const {
  HISTORICAL_VIEW
} = _redux.constants;
const socket = (0, _socket.default)(process.env.SERVER_URL);
const {
  updateItem
} = _redux.actions;
const {
  UPDATE_ITEM,
  UPDATE_ITEM_SUCCESS,
  UPDATE_ITEM_FAILED
} = _redux.constants;
const {
  deliveryStates
} = _redux.constants;
const {
  DELIVERED,
  CANCELLED
} = deliveryStates;
const inActiveStates = [DELIVERED, CANCELLED];

const Items = () => {
  const {
    dispatch,
    state: {
      items,
      updating,
      view,
      filter
    }
  } = (0, _react.useContext)(_redux.StoreContext);
  const [state, setState] = (0, _react.useState)({});

  const handleChange = id => e => {
    const {
      name,
      value
    } = e.target;
    setState(prevState => ({ ...prevState,
      [id]: {
        [name]: value
      }
    }));
  };

  const handleSubmit = id => e => {
    e.preventDefault();
    const stateData = state[id];
    if (!stateData) return null;
    const data = { ...stateData,
      id
    };

    const update = type => dispatch(updateItem(data, type));

    update(UPDATE_ITEM);
    socket.emit(`update`, data, err => {
      if (err) {
        return update(UPDATE_ITEM_FAILED);
      }

      update(UPDATE_ITEM_SUCCESS);
    });
  };

  const filteredItems = items.filter(({
    event_name
  }) => {
    const isInactive = inActiveStates.includes(event_name);

    switch (view) {
      case HISTORICAL_VIEW:
        return isInactive;
        break;

      default:
        return filter.length ? filter.includes(event_name) : !isInactive;
        break;
    }
  }); // Add some placeholder items for loading state

  if (filteredItems.length < 4) {
    for (let i = filteredItems.length; i < 4; i++) {
      filteredItems.push({
        loading: true
      });
    }
  }

  return _react.default.createElement(_rebass.Box, {
    width: 3 / 4
  }, _react.default.createElement(_ItemsNav.default, null), _react.default.createElement(_rebass.Box, {
    sx: {
      position: `relative`,
      borderStyle: `solid`,
      borderWidth: `3px`,
      borderTop: `0`,
      borderLeft: `0`,
      borderRight: `0`,
      borderColor: `secondary`
    }
  }, filteredItems.map(({
    event_name,
    destination,
    name,
    id,
    sent_at_second,
    loading
  }, i) => _react.default.createElement(_Item.default, {
    key: loading ? `loading-${i}` : `${id}-${sent_at_second}`,
    idx: i,
    eventName: event_name,
    name: state[id]?.name || name,
    bg: i % 2 === 0 ? `grey` : `white`,
    destination: state[id]?.destination || destination,
    loading: loading,
    view: view,
    updating: updating.includes(id),
    handleChange: handleChange(id),
    handleSubmit: handleSubmit(id)
  }))));
};

var _default = Items;
exports.default = _default;