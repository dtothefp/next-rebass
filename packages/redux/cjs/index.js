"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  StoreContext: true,
  createStore: true
};
exports.createStore = exports.StoreContext = void 0;

var _react = require("react");

var _store = require("./store");

Object.keys(_store).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _store[key];
    }
  });
});
const StoreContext = (0, _react.createContext)();
exports.StoreContext = StoreContext;

const createStore = (reducer, store = {}) => {
  const [state, dispatch] = (0, _react.useReducer)(reducer, store);
  return {
    state,
    dispatch
  };
};

exports.createStore = createStore;