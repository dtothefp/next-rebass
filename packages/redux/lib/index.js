import { createContext, useReducer } from 'react';

export * from './store';

export const StoreContext = createContext();

export const createStore = (reducer, store = {}) => {
  const [ state, dispatch ] = useReducer(reducer, store);

  return {state, dispatch};
};

