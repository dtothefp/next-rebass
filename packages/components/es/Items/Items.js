import React from "react";
import io from 'socket.io-client';
import { useState, useContext } from 'react';
import { actions, constants, StoreContext } from '@css/redux';
import { Box, Flex, Button } from 'rebass';
import Item from '../Item/Item';
const socket = io(process.env.SERVER_URL);
const {
  changeItemView,
  updateItem
} = actions;
const {
  UPDATE_ITEM,
  UPDATE_ITEM_SUCCESS,
  UPDATE_ITEM_FAILED
} = constants;
const {
  deliveryStates
} = constants;
const {
  DELIVERED,
  CANCELLED
} = deliveryStates;
const inActiveStates = [DELIVERED, CANCELLED];
export default (() => {
  const {
    dispatch,
    state: {
      items,
      updating,
      view,
      filter
    }
  } = useContext(StoreContext);
  const [state, setState] = useState({});

  const handleChange = id => e => {
    const {
      name,
      value
    } = e.target;
    setState(prevState => ({ ...prevState,
      [id]: {
        [name]: value
      }
    }));
  };

  const handleSubmit = id => e => {
    e.preventDefault();
    const stateData = state[id];
    if (!stateData) return null;
    const data = { ...stateData,
      id
    };

    const update = type => dispatch(updateItem(data, type));

    update(UPDATE_ITEM);
    socket.emit('update', data, err => {
      if (err) {
        return update(UPDATE_ITEM_FAILED);
      }

      update(UPDATE_ITEM_SUCCESS);
    });
  };

  const handleClick = view => () => {
    dispatch(changeItemView(view));
  };

  const filteredItems = items.filter(({
    event_name
  }) => {
    const isInactive = inActiveStates.includes(event_name);

    switch (view) {
      case "historical":
        return isInactive;
        break;

      default:
        return filter.length ? filter.includes(event_name) : !isInactive;
        break;
    }
  });
  const sx = {
    cursor: 'pointer'
  };
  return React.createElement(Box, {
    width: 3 / 4,
    height: "100vh"
  }, React.createElement(Flex, null, React.createElement(Button, {
    sx: sx,
    width: 1 / 2,
    variant: "primary",
    onClick: handleClick("active")
  }, "View Active"), React.createElement(Button, {
    sx: sx,
    width: 1 / 2,
    variant: "secondary",
    onClick: handleClick("historical")
  }, "Button")), filteredItems.map(({
    event_name,
    destination,
    name,
    id,
    sent_at_second
  }) => React.createElement(Item, {
    key: "".concat(id, "-").concat(sent_at_second),
    eventName: event_name,
    name: state[id]?.name || name,
    destination: state[id]?.destination || destination,
    disabled: updating.includes(id),
    handleChange: handleChange(id),
    handleSubmit: handleSubmit(id)
  })));
});