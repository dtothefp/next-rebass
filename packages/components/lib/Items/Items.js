import { useContext } from 'react';
import { StoreContext } from '@css/redux';
import Item from '../Item/Item';

export default () => {
  const {state} = useContext(StoreContext);

  return (
    <div>
      {state.items.map((item) => {
        const {name, id, sent_at_second: sent} = item;

        return (
          <Item
            {...item}
            key={id + name + sent}
          />
        );
      })}
    </div>
  );
};
