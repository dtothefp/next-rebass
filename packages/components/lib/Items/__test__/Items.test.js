import React from 'react';
import { mount, render } from 'enzyme';
import { constants, StoreContext } from '@css/redux';
import {socket as socketMock} from 'socket.io-client';
import Items from '../Items';


const {Provider} = StoreContext;
const {deliveryStates, ACTIVE_VIEW, HISTORICAL_VIEW, UPDATE_ITEM} = constants;
const {CREATED, DELIVERED} = deliveryStates;

describe(`#Items`, () => {
  const dispatch = jest.fn();
  const state = {
    items: [],
    updating: [],
    filter: [],
    view: ACTIVE_VIEW,
  };

  afterEach(() => {
    dispatch.mockClear();
    socketMock.mockClear();
  });

  it(`renders`, () => {
    expect(
      render(
        <Provider value={{dispatch, state}}>
          <Items />
        </Provider>
      )
    ).toMatchSnapshot();
  });

  it.skip(`renders 4 loading components if no items data is present`, () => {
    // TODO: can't figure out how to select Styled Components without pre-defined classnames
    // to measure that the length of items is 4
    const wrapper = mount(
      <Provider value={{dispatch, state}}>
        <Items />
      </Provider>
    );

    expect(wrapper.children().length).toBe(4);
  });

  it(`allows updating active items`, () => {
    const name = `name`;
    const destination = `destination`;
    const id = `ID`;
    const items = [{
      event_name: CREATED,
      destination,
      name,
      id,
      sent_at_second: 1,
    }];
    const newState = { ...state, items };
    const wrapper = mount(
      <Provider value={{dispatch, state: newState}}>
        <Items />
      </Provider>
    );

    const nameInput = wrapper.find(`input[name="${name}"]`);
    const destInput = wrapper.find(`input[name="${destination}"]`);
    const form = wrapper.find(`form`);

    nameInput.getDOMNode().value = `update-name`;
    nameInput.simulate(`change`);

    destInput.getDOMNode().value = `update-dest`;
    destInput.simulate(`change`);

    form.simulate(`submit`);

    const itemData = {
      name: `update-name`,
      destination: `update-dest`,
      id
    };

    expect(dispatch.mock.calls[0][0]).toEqual({
      type: UPDATE_ITEM,
      item: itemData
    });

    expect(socketMock.mock.calls[0][1]).toEqual(itemData);
  });

  it(`disables updating historical items`, () => {
    const name = `name`;
    const destination = `destination`;
    const id = `ID`;
    const view = HISTORICAL_VIEW;
    const items = [{
      event_name: DELIVERED,
      destination,
      name,
      id,
      sent_at_second: 1,
    }];
    const newState = {...state, items, view}
    const wrapper = mount(
      <Provider value={{dispatch, state: newState}}>
        <Items />
      </Provider>
    );

    const nameInput = wrapper.find(`input[name="${name}"]`).getDOMNode();
    const destInput = wrapper.find(`input[name="${destination}"]`).getDOMNode();

    expect(nameInput.disabled).toBeTruthy();
    expect(destInput.disabled).toBeTruthy();
  });
});
