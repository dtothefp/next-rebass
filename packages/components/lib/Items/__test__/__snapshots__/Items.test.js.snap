// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Items renders 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-1vrn7q0",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "css-112hoo",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "css-vysmdk",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "css-o49gr9",
                  "disabled": "",
                },
                "children": Array [
                  Object {
                    "data": "Active Orders",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "css-1t2d71z",
                  },
                  "children": Array [
                    Object {
                      "data": "Historical Orders",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "disabled": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "disabled": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "css-1t2d71z",
                },
                "children": Array [
                  Object {
                    "data": "Historical Orders",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "css-o49gr9",
                    "disabled": "",
                  },
                  "children": Array [
                    Object {
                      "data": "Active Orders",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "disabled": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "disabled": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "css-1tbpetn",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "sc-AykKC fhVZNx css-vurnku",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "sc-AykKC ljxIMJ css-vurnku",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "sc-AykKC fhVZNx css-vurnku",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "sc-AykKC ljxIMJ css-vurnku",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "sc-AykKC ljxIMJ css-vurnku",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "sc-AykKC fhVZNx css-vurnku",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "sc-AykKC ljxIMJ css-vurnku",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "sc-AykKC fhVZNx css-vurnku",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "sc-AykKC fhVZNx css-vurnku",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "sc-AykKC ljxIMJ css-vurnku",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "sc-AykKC ljxIMJ css-vurnku",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "sc-AykKC fhVZNx css-vurnku",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "sc-AykKC ljxIMJ css-vurnku",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "sc-AykKC fhVZNx css-vurnku",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "sc-AykKC ljxIMJ css-vurnku",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "sc-AykKC fhVZNx css-vurnku",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "css-1tbpetn",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "sc-AykKC fhVZNx css-vurnku",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "sc-AykKC ljxIMJ css-vurnku",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "sc-AykKC fhVZNx css-vurnku",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "sc-AykKC ljxIMJ css-vurnku",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "sc-AykKC ljxIMJ css-vurnku",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "sc-AykKC fhVZNx css-vurnku",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "sc-AykKC ljxIMJ css-vurnku",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "sc-AykKC fhVZNx css-vurnku",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "sc-AykKC fhVZNx css-vurnku",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "sc-AykKC ljxIMJ css-vurnku",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "sc-AykKC ljxIMJ css-vurnku",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "sc-AykKC fhVZNx css-vurnku",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "sc-AykKC ljxIMJ css-vurnku",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "sc-AykKC fhVZNx css-vurnku",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "sc-AykKC ljxIMJ css-vurnku",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "sc-AykKC fhVZNx css-vurnku",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "css-112hoo",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "css-vysmdk",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "css-o49gr9",
                    "disabled": "",
                  },
                  "children": Array [
                    Object {
                      "data": "Active Orders",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "css-1t2d71z",
                    },
                    "children": Array [
                      Object {
                        "data": "Historical Orders",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "button",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "disabled": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "disabled": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "class": "css-1t2d71z",
                  },
                  "children": Array [
                    Object {
                      "data": "Historical Orders",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "css-o49gr9",
                      "disabled": "",
                    },
                    "children": Array [
                      Object {
                        "data": "Active Orders",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "button",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "disabled": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "disabled": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
